source/renderer/app/analytics/TrackedRoute.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { FC, useEffect } from 'react';
+import { matchPath, Route, RouteProps } from 'react-router';
+import { useAnalytics } from '../components/analytics';
+
+export type TrackedRouteProps = RouteProps & {
+  pageTitle: string;
+};
+
+const TrackedRoute: FC<TrackedRouteProps> = (props) => {
+  const analytics = useAnalytics();
+  const { pageTitle, ...restProps } = props;
+
+  useEffect(() => {
+    const match = matchPath(window.location.hash.replace('#', ''), props);
+
+    if (match !== null) {
+      analytics.sendPageNavigationEvent(pageTitle);
+    }
+  }, [window.location.hash, props, pageTitle]);
```

Comment:
Yes, that will be covered by the 2nd after reorganizing changes on this branch.

Reply:
Actually, I reviewed this once again and that was done intentionally - since this component is a HOC for React-Router's Route component - potentially any of the route component props could change, but in general it is unlikely. This is why I put just `props` as a dependency. What I can do however - I can remove `pageTitle` from dependencies, because they are anyway part of `props`